### DISCUSSION TITLE
Change board name reported by serial

### DISCUSSION THREAD

**Discussion by Kamoool** [stm32duino/Arduino_Core_STM32#1755] [Q&A] [answered]
Hello,

Is it possible to change board name when connected by serial? Right now I'm getting "GENERIC_L432KBUX CDC in FS Mode", but I would like to change it to something custom.

**Comment by JiriBilek** [comment, marked as answer]
I believe the magic is in the file `....\stm32\2.3.0\cores\arduino\stm32\usb\usbd_desc.c`
I haven't tried it yet here in Arduino environment, but if you define `USB_PRODUCT_STRING `to your desired name it should work. The definition should be global, given as -D directive for the compiler. This can be done by creating/editing build_opt.h file (https://github.com/stm32duino/wiki/wiki/Customize-build-options-using-build_opt.h).

**Reply by fpistm** [reply]
Hi @Kamoool 

@JiriBilek answer's is right. Anyway, pay attention that the last release contains a regression as stated in the release note:

> ## Known issue
>  * #1749  build_opt.h not properly generated (see issue to get patch)

So think to apply the patch before using the build_opt.h feature.

**Comment by Kamoool** [comment]
Hey @JiriBilek, @fpistm,
I have upgraded to 2.3.0 and applied the patch. The description is now set correctly. One more thing - is there a way of changing the name reported in the windows device manager, or that's purely driver based?
With 2.3.0 upgrade one issue came up - EEPROM is not working. While uploading I see that message:
![image](https://user-images.githubusercontent.com/38004539/177143186-2aa0502a-2da0-4664-b6ce-b811b5f4b268.png)
That doesn't seem to be correct, because L432KBU6 doesn't have 512kB flash. I remember one issue back in 1.8.0 where Flash size was correct for KCU6, but wrong for KBU6. Unfortunatelly patch that was working back then can't be applied now, because of a different code. Can you help me with that?
I tried rolling back to 2.2.0, but it still doesn't work...

**Reply by fpistm** [reply]
Update the cube programmer. The reporter size should be correct. It is not link to the core.

**Reply by Kamoool** [reply]
Hmm.... Looking at the flash size and ram size reported by core it seems to be correct. I'll update the cube programmer and come back with the results.

**Reply by Kamoool** [reply]
That's what I'm seeing right now:
![image](https://user-images.githubusercontent.com/38004539/177145265-9db79f51-f317-41e2-a398-e3f61b93d653.png)

EEPROM is still not working though. Any ideas?

**Reply by Kamoool** [reply]
That's what I'm getting after the eeprom_put example:
![image](https://user-images.githubusercontent.com/38004539/177145997-65189c7a-38d0-445c-8f26-0c4227f7e052.png)
That's what I'm getting after the eeprom_get example:
![image](https://user-images.githubusercontent.com/38004539/177146097-c405d014-dc59-40fa-b256-1d6565bae342.png)

**Reply by Kamoool** [reply]
Tested on Nucleo32-L432KC and it works. Seems to be an issue with L432KB once again.
Results after L432KC test:
![image](https://user-images.githubusercontent.com/38004539/177146913-1b7aeaf4-b4e9-4847-903a-81dd00acfb0e.png)

**Comment by Kamoool** [comment]
As an answer to the EEPROM issue - that's an issue coming back since 1.8.0:
Resolution -[1316#issuecomment-839510855](https://github.com/stm32duino/Arduino_Core_STM32/issues/1316#issuecomment-839510855)
